sale_quotation_template_product_multi_add/wizards/sale_template_add_products.py

```diff
@@ -0,0 +1,73 @@
+from odoo import api, fields, models
+
+
+class SaleTemplateAddProducts(models.TransientModel):
+    _name = "sale.template.add.products"
+    _description = "Quotation Template Add Products"
+
+    product_ids = fields.Many2many(comodel_name="product.product")
+    item_ids = fields.One2many(
+        comodel_name="sale.template.add.products.item",
+        inverse_name="wizard_id",
+    )
+
+    def create_items(self):
+        for wizard in self:
+            vals_list = [
+                {"product_id": r.id, "wizard_id": wizard.id} for r in wizard.product_ids
+            ]
+            self.env["sale.template.add.products.item"].create(vals_list)
+        view = self.env.ref(
+            "sale_quotation_template_product_multi_add.view_add_products_to_qt2"
+        )
+        return {
+            "type": "ir.actions.act_window",
+            "res_model": self._name,
+            "view_type": "form",
+            "view_mode": "form",
+            "views": [(view.id, "form")],
+            "target": "new",
+            "res_id": self.id,
+            "context": self.env.context,
+        }
+
+    @api.model
+    def _get_line_values(self, qt, item):
+        sale_line = self.env["sale.order.template.line"].new(
```

Comment:
@ilyasProgrammer Why doing a new and convert to write ? Returning the dict is sufficent

Reply:
Right. Was copied from here: https://github.com/OCA/sale-workflow/blob/14.0/sale_product_multi_add/wizards/sale_import_products.py#L37

Reply:
That's because there was an onchange, so new values need to be added through the convert_to_write.

FYI, in that case I advice to use the onchange_helper module (you don't know all the possible onchanges) https://github.com/OCA/server-tools/tree/14.0/onchange_helper